Tests/ConfidenceTests/ConfidenceTest.swift

```diff
@@ -617,6 +617,29 @@ class ConfidenceTest: XCTestCase {
             XCTAssertEqual(error as? ConfidenceError, ConfidenceError.invalidContextInMessage)
         }
     }
+
+    func testConcurrentActivate() async {
+        for _ in 1...100 {
+            Task {
+                await concurrentActivate()
+            }
+        }
+    }
+
+    private func concurrentActivate() async {
+        let confidence = Confidence.Builder(clientSecret: "test")
```

Comment:
Don't we want to create a single instance and pass that to this function?

Reply:
`concurrentActivate` is meant to be the test we want to perform (it already has 10000 concurrent calls on the same instance). The wrapping `testConcurrentActivate` is just to be able to run the same test 100 times, since I noticed the exception is not always happening

Reply:
Yeah... and to be picky: running it more times is no _guarantee_ for it to happen.
Of course it would be amazing to come up with a guaranteed way to make it crash but if that's not possible I would be fine to merge this without the added test. Up to you 👍

Reply:
Agree: even with multiple runs, sometimes the test suite succeeds. Although it's worth mentioning the in the _majority_ of cases the repeated test does fail, at least on my machine.

Despite the imperfect testing, I think we are confident in the fix and we should get it out asap. I am going to merge and we improve the tests later